pkg/agent/agent.go

```diff
@@ -68,11 +69,25 @@ func (a *Agent) ID() string {
 	return a.peer.UID
 }
 
+func (a *Agent) sendControlMessage(msgFactory func(a string) (*messaging.Message, error)) {
+	msg, err := msgFactory(a.ID())
+	if err != nil {
+		a.logger.Error("failed to create control message", logging.Fields{"room": a.room, "peer": a.peer.UID, "error": err})
+	} else {
+		a.broker.Send(a.room, *msg)
+	}
+}
+
 // readPump handles messages coming from the peer
 func (a *Agent) readPump() {
+	a.sendControlMessage(messaging.NewPeerConnected)
 	a.broker.Register(a.room, a)
+
 	defer func() {
 		a.broker.Unregister(a.room, a)
+		a.sendControlMessage(messaging.NewPeerDisconnected)
+
+		close(a.stopChan)
 		a.conn.Close()
```

Comment:
We close the WS connection twice now (in reader and writer), probably the 2nd close returns an error. I think it'd be better to stick to the go rule that writers close channels, so can you remove it from here?